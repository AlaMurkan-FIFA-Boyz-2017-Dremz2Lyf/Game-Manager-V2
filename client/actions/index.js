// dependancies
import axios from 'axios';

// Action Types
import { SET_ERRORED, SET_LOADING, RECEIVE } from './types';

// Utilities
import { normalize } from '../utilities';

export const setErrored = (stateKey, data) => {
  // NOTE: Still not 100% sure if this is going to work for all errors.... but it might! :D
  let error = data.status ? {status: data.status, message:Â data.data} : {message: data.message, source: data.stack};

  return {
    type: SET_ERRORED,
    payload: {
      data: error,
      stateKey
    }
  };
};


export const setLoading = (stateKey, data) => {
  return {
    type: SET_LOADING,
    payload: {
      data,
      stateKey
    }
  };
};

export const receive = (stateKey, data) => {
  // receive is set up to accept an array or an object.
    // If it is an array we do the normalization here.
  if (Array.isArray(data)) {
    data = normalize(data);
  }

  // and send off our action with the proper payload
  return {
    type: RECEIVE,
    payload: {
      data,
      stateKey
    }
  };
};

export const fetch = (stateKey, data = {}) => {
  let request = axios.get(`/${stateKey}`, {params: data});

  return dispatch => {
    dispatch(setLoading(stateKey, true));

    return request.then(response => response.data).then(data => {
      dispatch(receive(stateKey, data));
    }).catch(error => {
      // This should handle any response errors, but more than likely not anything else.... D:
      let data = error.response ? error.response : error;
      dispatch(setErrored(stateKey, data));
    }).then(dis => {
      dispatch(setLoading(stateKey, false));
    });
  };
};

// In an effort to minimize request traffic, the server will respond with the updated or created data object we are currently working with. Instead of fetching all of that corresponding data we need after the post. We will just pass the updated data through to the receive action, by using getState from redux-thunk.

export const create = (stateKey, data) => {
  let post = axios.post(`/${stateKey}`, data);

  return (dispatch, getState) => {
    dispatch(setLoading(stateKey, true));

    return post.then(response => normalize(response.data)).then(updated => {

      // Grab the specific subState for the stateKey we are working with during this action
      let subState = getState().data[stateKey];

      // Take the subState and the updated data and create a new object
      let updatedSubState = {...subState, ...updated};

      // Pass that updated subState into the receive action
      return dispatch(receive(stateKey, updatedSubState));
    }).catch(error => {

      // If there was an error in the put, we grab the response from the server and pass it along to our setErrored action creator.
      let data = error.response;
      return dispatch(setErrored(stateKey, data));
    }).then(dis =>

      // Once everything is done, we set our loading flag to false.
      dispatch(setLoading(stateKey, false))
    );
  };
};

export const update = (stateKey, data) => {
  let put = axios.put(`/${stateKey}`, data);

  return (dispatch, getState) => {
    // Set our loading flag for the specific stateKey we are working with
    dispatch(setLoading(stateKey, true));

    // Then handle the async request.
    return put.then(response => normalize(response.data)).then(updated => {

      // Grab the specific subState for the stateKey we are working with during this action
      let subState = getState().data[stateKey];

      // Take the subState and the updated data and create a new object
      let updatedSubState = {...subState, ...updated};

      // Pass that updated subState into the receive action
      return dispatch(receive(stateKey, updatedSubState));
    }).catch(error => {

      // If there was an error in the put, we grab the response from the server and pass it along to our setErrored action creator.

      let data = error.response;
      return dispatch(setErrored(stateKey, data));
    }).then(dis =>

      // Once everything is done, we set our loading flag to false.
      dispatch(setLoading(stateKey, false))
    );
  };
};
