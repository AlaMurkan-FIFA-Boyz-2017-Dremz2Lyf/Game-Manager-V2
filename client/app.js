// Dependancies
import React from 'react';
import ReactDOM from 'react-dom';
import { ProviderÂ } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import promise from 'redux-promise';

// Components
import Home from './components/home';

import rootReducer from './reducers/index.js';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(rootReducer, undefined, composeEnhancers(
  applyMiddleware(promise)
));

ReactDOM.render(
  <Provider store={store}>
    <Home/>
  </Provider>,
  document.getElementById('app')
);
