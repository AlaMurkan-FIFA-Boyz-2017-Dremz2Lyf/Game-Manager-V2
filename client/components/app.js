// Dependancies
import React, { Component } from 'react';
import { Link } from 'react-router';
import {
  PageHeader,
  Grid
} from 'react-bootstrap';


class App extends Component {

  render() {
    return (
      <div className="home">
        <Link to='/'><PageHeader>Game Manager</PageHeader></Link>
        <Grid>
          {this.props.children}
        </Grid>
      </div>
    );
  }
}



export default App;
