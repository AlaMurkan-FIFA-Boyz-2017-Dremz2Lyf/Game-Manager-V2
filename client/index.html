<html>
<head>
  <title>Game Manager</title>
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans|Oswald|PT+Sans">
</head>
<body>
  <div id="app"></div>
</body>
</html>
