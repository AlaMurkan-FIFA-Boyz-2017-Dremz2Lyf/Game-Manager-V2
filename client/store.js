
// Make our store exportable for using in tests/and production

import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

import rootReducer from './reducers/index.js';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(rootReducer, undefined, composeEnhancers(
  applyMiddleware(thunk)
));
