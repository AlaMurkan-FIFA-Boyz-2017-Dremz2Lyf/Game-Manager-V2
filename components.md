 Component | State | Actions
---|---|---
 Home | None? | None?
 AllTimeStats | {currentList: ongoing/finished/new} | None?
