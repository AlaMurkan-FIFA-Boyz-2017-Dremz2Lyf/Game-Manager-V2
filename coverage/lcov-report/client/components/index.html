<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> client/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>95/95</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>25/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>51/51</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>91/91</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="create_player.js"><a href="create_player.js.html">create_player.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="create_tournament.js"><a href="create_tournament.js.html">create_tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	</tr>

<tr>
	<td class="file high" data-value="game.js"><a href="game.js.html">game.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file high" data-value="games.js"><a href="games.js.html">games.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="home.js"><a href="home.js.html">home.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="play_tournament.js"><a href="play_tournament.js.html">play_tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="player_form.js"><a href="player_form.js.html">player_form.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	</tr>

<tr>
	<td class="file high" data-value="player_list.js"><a href="player_list.js.html">player_list.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file high" data-value="player_list_item.js"><a href="player_list_item.js.html">player_list_item.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="tournament.js"><a href="tournament.js.html">tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="tournaments.js"><a href="tournaments.js.html">tournaments.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Mar 10 2017 17:46:58 GMT-0600 (CST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
