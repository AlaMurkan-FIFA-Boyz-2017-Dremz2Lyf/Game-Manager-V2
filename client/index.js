// Dependancies
import React from 'react';
import ReactDOM from 'react-dom';
import { ProviderÂ } from 'react-redux';

// css
import '../styles/styles.css';

// Bring in our router
import Routes from './router';

// Grab our root reducer and our store from their files
import rootReducer from './reducers/index.js';
import { store } from './store';

// Found this... Needed for hot module reloading.
if (module.hot) {
  module.hot.accept();
}

ReactDOM.render(
  <Provider store={store}>
    <Routes />
  </Provider>,
  document.getElementById('app')
);
