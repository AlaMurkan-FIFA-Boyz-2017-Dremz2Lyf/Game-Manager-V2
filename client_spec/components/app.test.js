import React from 'react';
import { Provider } from 'react-redux';

import { store } from '../../client/store';

import { App } from '../../client/components/app';

describe('Shallow App', () => {
  let wrapper = shallow(<App/>);

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call fetch when mounted', () => {
    let mockFetch = jest.fn();
    let mounted = mount(
      <Provider store={store}>
        <App
          fetch={mockFetch}
        />
      </Provider>
    );
    expect(mockFetch).toHaveBeenCalled();
    expect(mockFetch.mock.calls[0]).toEqual(['players']);
    mockFetch.mockClear();
  });
});
