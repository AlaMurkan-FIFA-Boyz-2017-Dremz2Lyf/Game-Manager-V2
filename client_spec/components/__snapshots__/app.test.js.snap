exports[`Shallow App should match the snapshot 1`] = `
<div
  className="home">
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/">
    <PageHeader
      bsClass="page-header">
      Game Manager
    </PageHeader>
  </Link>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false} />
</div>
`;
